cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -37,15 +37,22 @@ func Run(gadgetRoot, device string, options *Options) error {
 		return fmt.Errorf("cannot use empty device node")
 	}
 
-	// XXX: ensure we test that the current partition table is
-	//      compatible with the gadget
 	lv, err := gadget.PositionedVolumeFromGadget(gadgetRoot)
 	if err != nil {
 		return fmt.Errorf("cannot layout the volume: %v", err)
 	}
 
-	sfdisk := partition.NewSFDisk(device)
-	created, err := sfdisk.Create(lv)
+	diskLayout, err := partition.NewDeviceLayout(device)
+	if err != nil {
+		return fmt.Errorf("cannot read %v partitions: %v", device, err)
+	}
+
+	// check if the current partition table is compatible with the gadget
+	if err := ensureLayoutCompatibility(lv, diskLayout); err != nil {
```

Comment:
I wonder if ensureLayoutCompatibility() should be something we call inside diskLayout.Create(lv) - mostly for API robustness, i.e. I think we only want to run "Create()" if the layouts are compatible so it seems to make sense to automatically call it, WDYT?

Reply:
Hmm, I see your point and it makes sense from the robustness point of view, but conceptually the reason I left it outside was to keep a policy/mechanism-style separation in such a way that `partition.DeviceLayout` takes care of the mechanics of partition creation, while the higher-level `bootstrap.Run()` decides if the partition should be created or not. If you feel strongly inclined to the robustness side we can break this separation, otherwise I feel the internal organization could benefit from having those in different layers.

Reply:
That's fine, let's keep it this way then. I would love to hear @pedronis  opinion (just our of curiosity, not because I feel strong either way) but he is on vac. One we get a second :+1: (maybe from @bboozzoo ?) this should land and we should move forward.